src/main/java/com/okta/tools/helpers/RoleHelper.java

```diff
@@ -121,7 +121,7 @@ public AssumeRoleWithSAMLRequest chooseAwsRoleToAssume(String samlResponse) thro
     public List<AccountOption> getAvailableRoles(String samlResponse) throws IOException {
         Map<String, String> roles = AwsSamlRoleUtils.getRoles(samlResponse);
         if (roles.size() == 1) {
-            String roleArn = roles.keySet().iterator().next();
+            String roleArn = roles.values().iterator().next();
```

Comment:
I think this is the wrong order, I had to change it to `keySet` in https://github.com/oktadeveloper/okta-aws-cli-assume-role/pull/318/files#diff-ce84c5de3dda5feab65c23337eb4306dR124

Can you explain more why this map should be keyed off principalArn instead of roleArn? Consider the case where you have multiple different roles under the same principal.

Reply:
What I remember is that the correct role string used from the role selection list was the actual value and not they key in that map:
here's the actual commit:
https://sourcegraph.com/github.com/oktadeveloper/okta-aws-cli-assume-role/-/commit/abaf2c1e0083217665135d4948a7542ba34a2b26